db/models.py

```diff
@@ -46,3 +54,73 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return self.created_at.strftime("%Y-%m-%d %H:%M:%S")
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey("MovieSession", on_delete=models.CASCADE)
+    order = models.ForeignKey(to="Order", on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["row", "seat", "movie_session"],
+                name="unique_row_seat_movie_session"
+            )
+        ]
+
+    def __str__(self) -> str:
+        return f"{self.movie_session.movie.title} " \
+               f"{self.movie_session.show_time} " \
+               f"(row: {self.row}, seat: {self.seat})"
+
+    def clean(self) -> None:
+        rows = self.movie_session.cinema_hall.rows
+        if not (1 <= self.row <= rows):
+            raise ValidationError(
+                {
+                    "row": ["row number must be in available range: "
+                            f"(1, rows): (1, {rows})"]
+                }
+            )
+
+        seats = self.movie_session.cinema_hall.seats_in_row
+        if not (1 <= self.seat <= seats):
+            raise ValidationError(
+                {
+                    "seat": ["seat number must be in available range: "
+                             f"(1, seats_in_row): (1, {seats})"]
+                }
+            )
+
+    def save(self, force_insert: bool = False,
+             force_update: bool = False,
+             using: str = None,
+             update_fields: str = None
+             ) -> None:
```

Comment:
Fix indentation. And update_field is a list, also you use a default value None, so use annotation for both variants. From python 3.10 `update_fields: list | None` if lower `updated_fields: Optional[list] = None`

Reply:
The same about using, use optional annotation for str